app/services/sidebar_site_settings_backfiller.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+# A service class that backfills the changes to the default sidebar categories and tags site settings.
+#
+# When a category/tag is removed from the site settings, the `SidebarSectionLink` records associated with the category/tag
+# are deleted.
+#
+# When a category/tag is added to the site settings, a `SidebarSectionLink` record for the associated category/tag are
+# created for all users that do not already have a `SidebarSectionLink` record for the category/tag.
+class SidebarSiteSettingsBackfiller
+  def initialize(setting_name, previous_value:, new_value:)
+    @setting_name = setting_name
+
+    @linkable_klass, previous_ids, new_ids =
+      case setting_name
+      when "default_sidebar_categories"
+        [
+          Category,
+          previous_value.split("|"),
+          new_value.split("|")
+        ]
+      when "default_sidebar_tags"
+        klass = Tag
+
+        [
+          klass,
+          klass.where(name: previous_value.split("|")).pluck(:id),
+          klass.where(name: new_value.split("|")).pluck(:id)
+        ]
+      end
```

Comment:
Nitpick/personal opinion, but I'd add an `else` condition here that raises an error if the setting name doesn't match anything because it'd make it immediately clear what's wrong when e.g. someone makes a typo when using this class.

Reply:
Good point, I've made the changes.